### PR TITLE
Fixes #9997 - Release process docs

### PR DESCRIPTION
**Pull request by eLobato**
cc @witlessbird @daviddavis @parthaa please take a look

### CODE REVIEW

File: release/RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+* Bump `lib/foreman_docker/version.rb` according to the Versioning policy specified in this document.
+* Run `changelog` and update the `Contributors` and `CHANGELOG` files with those generated by the
+script
+* `git add` your changes
+* `git commit -m "Bump version to x.y.z"`
+* `git push upstream master`
+* `git tag x.y.z`
+* `git push upstream x.y.z`
+* `git checkout x.0-stable`
+* `git pull --rebase upstream master`
```
Comment (domcleal) [inline release/RELEASE.md]: rebasing a stable branch on master?  Are you sure?

Comment (eLobato) [inline release/RELEASE.md]: Currently our releases are essentially master snapshots

Comment (domcleal) [inline release/RELEASE.md]: ah okay, I'd perhaps suggest removing --rebase or using `git merge --ff-only master` then so you don't accidentally rebase the stable branch if something _does_ change

**Comment by daviddavis** [comment]
Seems like the release process could be automated in a script.

File: release/RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+## Tips before releasing
+
+* Test against the current Foreman version. For instance, the person releasing 1.0.2, should
+ensure it works well with Foreman 1.7 as that is the version supported by 1.y.z releases.
+* Always test using Foreman and Katello, as Katello will depend on `foreman-docker` for some tasks.
+Use the latest Katello version that is stable with the Foreman version supported.
+* Make sure there are no discrepancies in the dependencies specified in the gemspec versus the
+dependencies of the RPM spec and deb sources.
+* If any dependency in the gemspec was updated, and the package in foreman-packaging does not serve
+it, update the package or consider dropping the dependency.
```
Comment (daviddavis) [inline release/RELEASE.md]: Is this just a manual check?

Comment (domcleal) [inline release/RELEASE.md]: I recommend using gem-compare.  https://github.com/theforeman/foreman-packaging/tree/rpm/develop#howto-update-a-package

Comment (daviddavis) [inline release/RELEASE.md]: :+1: 

@eLobato can we make a note about gem-compare?

Comment (domcleal) [inline release/RELEASE.md]: Actually, re-reading the context gem-compare doesn't make a lot of sense, this is prior to release.

File: release/RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+These sanity checks should become less and less necessary as end-to-end tests improve and integration tests are added.
+
+## Process
+
+* Test the master branch according to the tips mentioned above
+* Bump `lib/foreman_docker/version.rb` according to the Versioning policy specified in this document.
+* Run `changelog` and update the `Contributors` and `CHANGELOG` files with those generated by the
+script
+* `git add` your changes
```
Comment (daviddavis) [inline release/RELEASE.md]: It's probably easier to escape this whole block with \`\`\` instead of each line.

**Comment by daviddavis** [comment]
I went through the docs last week to release 1.2.3. Worked well. :+1: from me.

**Comment by daviddavis** [comment]
Wondering if we should attach a redmine issue though.

**Comment by daviddavis** [comment]
[test]

**Comment by daviddavis** [comment]
[test]

**Comment by eLobato** [comment]
@daviddavis http://projects.theforeman.org/issues/9997 :smile:

**Comment by daviddavis** [comment]
ACK

**Comment by daviddavis** [comment]
Merged as 741cff160f273936fcaa19d83276018cf0661aef

**Pull request closed without merging**